fcli-core/fcli-ssc/src/main/java/com/fortify/cli/ssc/appversion/cli/cmd/SSCAppVersionCommands.java

```diff
@@ -23,6 +23,7 @@
         subcommands = {
             SSCAppVersionCreateCommand.class,
             SSCAppVersionDeleteCommand.class,
+            SSCAppVersionCopyStateCommand.class,
```

Comment:
what if you want to get the issues from feat/feature-1 to dev, then feat/feature-2  to dev, and so on

Reply:
I don't think this will work in a reliable way. For example:
- Create branches `feat1` and `feat2` from current `dev` branch, containing 5 critical issues
- Fix 2 critical issues in `feat1`
- Merge `feat1` into `dev`, effectively having 3 critical issues in `dev` and `feat1`, but still 5 in `feat2`
- Merge `feat2` into `dev`, 'overriding' current dev state with `feat2` state, causing those 2 critical issues to show up as 're-introduced' in SSC

Potentially, these 2 issues would be marked as 'removed' again when running a new scan on `dev`, but it will cause weird/incorrect issue history & variable/performance indicator values. There are probably a lot more examples where this wouldn't work reliably.

Reply:
Hi @rsenden, to add to the use cases that we are looking for.

In our git workflow we have the develop branch for snapshots and the main branch for releases, however developers are complaining constantly that they already suppress the issues in develop and they do not want to do double work and also suppress them in the master branch... that is the main use case we want to solve with this command.

Devs only work on the develop branch for triage and audit of issues (suppress false positives) , but when they use master branch our idea is to copy the state always of develop into the master branch, before running a new fortify sca scan on master. That way they do not have to triage again the issues and can see the impact their code changes will have on the issues, it may create more or it may resolve.

Things get a bit more complicated once we add feature and hot fix branches to the mix, this fcli copy could really help to solve that issue by alway copying first the develop branch state (effectvily the only branch developers work on) and then running the scan and having the results sorted by introduced date.

We also wanted to do a diff of only the issues when doing PR from feature branch into develop branch but did not find a good way to do this.

In essence what this feature could solve, is to mark an issue suppressed across all application branches so that security gates can be checked across all branches and developers do not have to do double or triple the work. It would also be a great feature to have on the UI. This is currently the number one complaint we get from our devs using Fortify SSC.

Reply:
Hi @webmutation, thanks for the additional input. I agree this would be very useful functionality, but unfortunately I don't think such use cases are currently properly supported by SSC. As explained in my previous comment, it may cause issues to continuously jump between 'removed', 'newly introduced', 're-introduced', ... states, potentially affecting corresponding issue introduced/removed/... dates and variable & performance indicator values.

Maybe it's possible to come up with a working solution by following a very strict process and ordering of git merges and SSC application version state copying/merging, but I think this would be fragile. So, before adding this command to fcli, we first need to discuss any potential implications with the SSC engineering team. Worst case, SSC enhancements are needed to properly support such use cases, before we can add support for this fcli. Have you already submitted a product enhancement request through our regular support channels for having this functionality added to SSC?

Reply:
Thanks for your input @rsenden, indeed it is fragile, we experience already some of the effects you mentioned, namely the issue introduction date of an issue on the PR branch is not the same as in the develop branch... the only way we found to make it stable was indeed to have a strict git process. We are currently using REST calls for this, we will then proceed with that method until this command is added to fcli or something along these lines/

I am on a different team than Alejo here on this thread, but we work for the same organization (EC), his unit is the one that is providing us with Fortify as a service. I will request them to open a product enhancement via the support channel.

Reply:
Hi @webmutation, can you please provide a high-level overview of the process steps that you have in place with regards to git merges, Fortify scans and SSC state copying? I think it would be useful to provide this information in the description of the fcli `copy-state` command.

As for the issue introduction date not being copied, I've been told that the upcoming SSC 23.2 release will copy this as well.